### ISSUE TITLE
Update documentation on how to retrieve an object from a key. 

### ISSUE THREAD

**Issue by jaswanthm** [florimondmanca/djangorestframework-api-key#144]
Existing documentation

```
from rest_framework.views import APIView
from rest_framework_api_key.models import APIKey
from rest_framework_api_key.permissions import HasAPIKey

from .models import Project

class ProjectListView(APIView):
    permission_classes = [HasAPIKey]

    def get(self, request):
        """Retrieve a project based on the request API key."""
        key = request.META["HTTP_AUTHORIZATION"]
        api_key = APIKey.objects.get_from_key(key)
        project = Project.objects.get(api_key=api_key)
```

**Issue** :  

`key = request.META["HTTP_AUTHORIZATION"]` returns a string in the following format :  `ApiKey *****`

One of the ways this can be solved is by 

`key = request.META["HTTP_AUTHORIZATION"].split()[1]` 

Happy to get some advice on other ways to do this as well.

**Comment by florimondmanca** [comment]
Hello — you're absolutely right, good catch!

I see two options for us here:

* Use the default assumption that the key comes from the `Authorization` header, and do as you said:

```python
key = request.META["HTTP_AUTHORIZATION"].split()[1]
```

* Use the [key parser](https://florimondmanca.github.io/djangorestframework-api-key/guide/#api-key-parsing) stored on the permission class:

```python
key = HasAPIKey.key_parser.get(request)
```

* Use the `.get_key()` method of the permission class (right now feels a bit awkward because we have to build a throw-away instance of the permission class, since `.get_key()` is an instance method):

```python
key = HasAPIKey().get_key(request)
```

While the last two options might seem more general, there are subtle potential gotchas. For example, the permission classes must match what's on the view.

From this I'd guess we could just fix things for the default case for now, and do as you suggested? Any extra hints on handling custom permission classes/key parsing could be treated as follow ups.

**Comment by jaswanthm** [comment]
Hey!

> From this I'd guess we could just fix things for the default case for now. 

Totally agree with this. It would be hard to provide documentation for multiple aspects. Having one working example for the default scenario should give everyone an understanding of how it works. 

Would you like me to create a pull request or are you happy to make that change ? Let me know. Ta

**Comment by florimondmanca** [comment]
@jaswanthm Sure, pull requests welcome!

**Comment by florimondmanca** [comment]
Closed via #149.
